codes/ruby/chapter_stack_and_queue/queue.rb

```diff
@@ -0,0 +1,37 @@
+=begin
+File: queue.rb
+Created Time: 2024-04-06
+Author: Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
+=end
+
+### Driver Code ###
+
+# 初始化队列
+# Ruby 内置的队列 （Thread::Queue) 没有 peek 和遍历方法，可以把 Array 当作队列来使用
+queue = []
+
+# 元素入队
+queue.push(1)
+queue.push(3)
+queue.push(2)
+queue.push(5)
+queue.push(4)
+puts "队列 queue = #{queue}"
+
+# 访问队列元素
+peek = queue.first
+puts "队首元素 peek = #{peek}"
+
+# 元素出队
+# 清注意，由于是数组，Array#shift 方法时间复杂度为 O(n)
```

Comment:
Is `Array#shift` a standard way to denote the class method?

Reply:
It is a standard way to denote the instance method in Ruby.
In Ruby API documentation, the `Class#method` format represents the instance method and the `Class.method` represents the class method (static method)

For example:
https://www.rubydoc.info/stdlib/core/Array:shift